src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,126 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_ARRAY_LIST_CAPACITY = 10;
+    private static final double ARRAY_MULTIPLIER = 1.5;
+    private T[] elements;
+    private int size;
+
+    public ArrayList() {
+        this.elements = (T[]) new Object[DEFAULT_ARRAY_LIST_CAPACITY];
+        this.size = 0;
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == elements.length) {
+            resize();
+        }
+        elements[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        indexOutOfRangeForAdd(index);
```

Comment:
You can add check `if (index == size) ` and then use `indexOutOfRange(index);` method

Reply:
Sry I can't get it:
1. I don't understand at what line you want this change.
2. I don't get the logic behind it.

Can you give me more details pls.

Reply:
It was meant to simply add a check so as not to call the method once again. But as for me, and so excellent